server/src/main/java/org/opensearch/cluster/routing/allocation/DiskThresholdMonitor.java

```diff
@@ -406,6 +415,27 @@ private void setLastRunTimeMillis() {
         lastRunTimeMillis.getAndUpdate(l -> Math.max(l, currentTimeMillisSupplier.getAsLong()));
     }
 
+    protected void setIndexCreateBlock(final ActionListener<Void> listener, boolean indexCreateBlock) {
+        final ActionListener<Void> wrappedListener = ActionListener.wrap(r -> {
+            setLastRunTimeMillis();
```

Comment:
this is only meant for reroute operations as it needs to honor `diskThresholdSettings.getRerouteInterval().millis()` . We are doing this in `updateIndicesReadOnly` , which we should remove as well .

Reply:
It's not just the internal, but a combination of interval plus the need to reroute. The `setLastRunTimeMillis` aims to skip reroute if it doesn't see a need to reroute even though the interval has elapsed.